tests/Integration/Behaviour/Features/Context/Domain/StateFeatureContext.php

```diff
@@ -56,6 +66,71 @@ public function __construct()
         $this->defaultLangId = $configuration->get('PS_LANG_DEFAULT');
     }
 
+    /**
+     * @When I add new state :stateReference with following properties:
+     *
+     * @param string $stateReference
+     * @param TableNode $table
+     */
+    public function createState(string $stateReference, TableNode $table): void
+    {
+        $data = $table->getRowsHash();
+
+        try {
+            /** @var StateId $stateId */
+            $stateId = $this->getCommandBus()->handle(new AddStateCommand(
+                (int) Country::getIdByName($this->defaultLangId, $data['country']),
+                Zone::getIdByName($data['zone']),
+                $data['name'],
+                $data['iso_code'],
+                PrimitiveUtils::castStringBooleanIntoBoolean($data['enabled'])
+            ));
+
+            SharedStorage::getStorage()->set($stateReference, new State($stateId->getValue()));
```

Comment:
```suggestion
            $this->getStorage()->set($stateReference, $stateId->getValue());
```

Reply:
Usually, we only store the IDs in the storage not full objects Besides even in the edit endpoint where you get it you only use `$state->id`

Reply:
You're right. It's fixed.